module3/homework/passwordCheck/validation.cpp

```diff
@@ -1,2 +1,86 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+bool czy_zawiera_cyfre(char znak)
+{
+    int cyfra = (int)znak;
+    return (cyfra >= 48) && (cyfra <= 57);
+}
+
+bool czy_zawiera_znak_specjalny(char znak)
+{
+    int cyfra = (int)znak;
+    return ((cyfra >= 0) && (cyfra <= 47)
+        || (cyfra >= 58) && (cyfra <= 64)
+        || (cyfra >= 91) && (cyfra <= 96)
+        || (cyfra >= 0) && (cyfra <= 47)
+        || (cyfra >= 123) && (cyfra <= 255));
+}
+
+bool czy_zawiera_wielka_litere(char znak)
+{
+    int cyfra = (int)znak;
+    return (cyfra >= 65) && (cyfra <= 90);
+}
+
+ErrorCode checkPasswordRules(std::string password)
+{
+    if (password.size() < 9) 
+        return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+
+    if (!std::any_of(password.begin(), password.end(), czy_zawiera_cyfre))
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+
+    if (!std::any_of(password.begin(), password.end(), czy_zawiera_znak_specjalny))
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    
+    if (!std::any_of(password.begin(), password.end(), czy_zawiera_wielka_litere))
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+
+    return ErrorCode::Ok;
+}
+
+
+
+ErrorCode checkPassword(std::string password, std::string repeatedPassword)
+{
+    if (doesPasswordsMatch(password, repeatedPassword))
+        return checkPasswordRules(password);
+    else 
+        return ErrorCode::PasswordsDoesNotMatch;
+   
+}
```

Comment:
You don't modify string so pass string by const reference
```suggestion
ErrorCode checkPassword(const std::string password&, const std::string& repeatedPassword)
{
    if (doesPasswordsMatch(password, repeatedPassword))
        return checkPasswordRules(password);
    else 
        return ErrorCode::PasswordsDoesNotMatch;
   
}
```